system-test/pubsub.ts

```diff
@@ -45,22 +59,67 @@ describe('pubsub', () => {
 
   const TOPIC_FULL_NAMES = TOPICS.map(getTopicName);
 
+  function generateName(name: string) {
+    return [PREFIX_WITH_RUNID, name, shortUUID(), CURRENT_TIME].join('-');
+  }
+
   function generateSnapshotName() {
-    return 'test-snapshot-' + uuid.v4();
+    return generateName('snapshot');
   }
 
   function generateSubName() {
-    return 'test-subscription-' + uuid.v4();
+    return generateName('subscription');
   }
 
   function generateTopicName() {
-    return 'test-topic-' + uuid.v4();
+    return generateName('topic');
   }
 
   function getTopicName(topic: Topic) {
     return topic.name.split('/').pop();
   }
 
+  function deleteTestResource(resource: Resource) {
+    // Delete resource from current test run.
+    if (resource.name.includes(PREFIX_WITH_RUNID)) {
```

Comment:
Seems like we don't need both the ID logic and the timestamp logic, I would vote we just do the timestamp.

Reply:
Removed run id and updated it with current time value with with milliseconds.